### PR TITLE
Use the new registry and mapper API of Legate

### PR DESCRIPTION
**Pull request by madsbk**
See https://github.com/nv-legate/legate.core/pull/675 and https://github.com/nv-legate/legate.core/pull/676

### CODE REVIEW

File: legate/cpp/legate_mapping.cpp [hunk 1]
```diff
@@ -81,15 +81,11 @@ Legion::Logger log_legate_kvikio(library_name);
 void registration_callback()
 {
   legate::ResourceConfig config;
-  config.max_mappers = 1;
-  config.max_tasks   = OP_NUM_TASK_IDS;
-
-  legate::LibraryContext context(library_name, config);
+  config.max_tasks = OP_NUM_TASK_IDS;
```
Comment (wence-) [inline legate/cpp/legate_mapping.cpp:84]:
```c++
legate::ResourceConfig config = {.max_tasks = OP_NUM_TASK_IDS};
```
?
Or are you worried about the struct gaining a constructor somewhere down the line?

**Review by wence- — APPROVED** [review]
Looks fine I think, possibly one minor style cleanup.

File: legate/cpp/legate_mapping.cpp (hunk 1 above)
Comment (madsbk) [reply legate/cpp/legate_mapping.cpp]: fixed

**Comment by madsbk** [comment]
Thanks @wence-, let me re-build and re-run the tests and then it can be merged

**Pull request merged**